### PR TITLE
Add ConfigProvider

### PR DESCRIPTION
**Pull request by geerteltink**
Adds ConfigProvider with basic configuration.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -51,6 +51,9 @@
             "dev-master": "2.1.x-dev",
             "dev-develop": "2.2.x-dev",
             "dev-release-3.0.0": "3.0.x-dev"
+        },
+        "zf": {
+            "config-provider": "Zend\\Expressive\\Router\\ConfigProvider"
```
Comment (michalbundyra) [inline composer.json:56]: Again, I would consider changing namespace

File: src/ConfigProvider.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-zendrouter for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
Comment (michalbundyra) [inline src/ConfigProvider.php]: only 2018 please for new files

File: src/ConfigProvider.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * @license   https://github.com/zendframework/zend-expressive-zendrouter/blob/master/LICENSE.md New BSD License
+ */
+
```
Comment (webimpress) [inline src/ConfigProvider.php]: remove empty line before strict type declaration

File: src/ConfigProvider.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        return [
+            'dependencies' => $this->getDependencies(),
+        ];
+    }
+
+    public function getDependencies() : array
+    {
+        return [
+            'factories' => [
```
Comment (michalbundyra) [inline src/ConfigProvider.php]: `invokables` ?
https://github.com/zendframework/zend-expressive-skeleton/blob/release-3.0.0/src/ExpressiveInstaller/Resources/config/router-zend-router.php#L10

I would consider to have here `invokable` for `ZendRouter`, and `aliases` for `RouterInterface`.

File: test/ConfigProviderTest.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-zendrouter for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
Comment (michalbundyra) [inline test/ConfigProviderTest.php]: 2018 only for new files, please

File: test/ConfigProviderTest.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ * @license   https://github.com/zendframework/zend-expressive-zendrouter/blob/master/LICENSE.md New BSD License
+ */
+
```
Comment (webimpress) [inline test/ConfigProviderTest.php]: no new line before strict_type declaration

File: test/ConfigProviderTest.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    protected function setUp() : void
+    {
+        $this->provider = new ConfigProvider();
+    }
+
+    public function testInvocationReturnsArray() : array
+    {
+        $config = ($this->provider)();
+        self::assertInternalType('array', $config);
```
Comment (michalbundyra) [inline test/ConfigProviderTest.php]: `$this->`... ?

File: test/ConfigProviderTest.php [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * @depends testInvocationReturnsArray
+     */
+    public function testReturnedArrayContainsDependencies(array $config) : void
+    {
+        self::assertArrayHasKey('dependencies', $config);
+        self::assertInternalType('array', $config['dependencies']);
+        self::assertArrayHasKey('factories', $config['dependencies']);
+        self::assertInternalType('array', $config['dependencies']['factories']);
+        self::assertArrayHasKey(RouterInterface::class, $config['dependencies']['factories']);
```
Comment (michalbundyra) [inline test/ConfigProviderTest.php]: `$this->` ... 5x ?

**Review by michalbundyra — CHANGES_REQUESTED** [review]

**Comment by weierophinney** [comment]
@xtreamwayz Same comments from the fastroute package apply here:

- Use a subnamespace for the `ConfigProvider`.
- Use `$this->` instead of `self::` for now.

**Comment by weierophinney** [comment]
Thanks, @xtreamwayz!

**Pull request merged**